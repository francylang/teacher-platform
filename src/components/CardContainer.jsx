import React, { Component } from 'react';
import DiscussionCard from './DiscussionCard';

class CardContainer extends Component {
  grabCommentsForDisscussion(id) {
    this.props.comments.filter(comment => {
      if (comment.discussionId === id) {
        return comment
      }
    })
  }

  buildDiscussionCard() {
    return this.props.discussions.map(discussion => {
      return (
        <DiscussionCard
          key={discussion.id}
          discussion={discussion}
          comment={this.grabCommentsForDisscussion(discussion.id)}
        />)
    })
  }

  render() {
    return (
      <div className="card-section">
        <h2 className="current-feed-title">Discussions:</h2>
        {this.buildDiscussionCard()}
      </div>
    );
  }
}

export default CardContainer;
