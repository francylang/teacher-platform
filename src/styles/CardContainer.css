.card-section {
  width: 65%; }

.discussions-heading {
  font-family: 'Ubuntu Mono';
  font-size: 160%; }
