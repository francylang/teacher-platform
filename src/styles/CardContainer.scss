.card-section {
  width: 65%;
}
