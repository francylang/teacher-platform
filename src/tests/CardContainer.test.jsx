import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from '../components/CardContainer';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<CardContainer />', () => {
  let wrapper;
  let mockFn = jest.fn();
  let filteredDiscussions = [
    {
      id: 1,
      tagTitle: "6.RP.A.1",
      discussions: [
        {
          tagTitle: "6.RP.A.1",
          title: "Unit Rate",
          body: "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
        },
      ],
    },
  ];
  let comments = [
    {
      comments: [
        {
          body: "Yes but it's important",
        },
        {
          body: "Here is a link to the progression docs: LINK",
        },
      ],
    }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <CardContainer
        filteredDiscussions={filteredDiscussions}
        comments={comments}
      />
    );
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a card section', () => {
    const card = wrapper.find('.card-section');

    expect(card.length).toEqual(1);
  });

  it('should render a card section title', () => {
    const title = wrapper.find('.discussions-heading');

    expect(title.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
