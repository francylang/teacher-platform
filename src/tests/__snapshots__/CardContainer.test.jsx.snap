// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContainer /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    comments={
        Array [
            Object {
              "comments": Array [
                Object {
                  "body": "Yes but it's important",
                },
                Object {
                  "body": "Here is a link to the progression docs: LINK",
                },
              ],
            },
          ]
    }
    filteredDiscussions={
        Array [
            Object {
              "discussions": Array [
                Object {
                  "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
                  "tagTitle": "6.RP.A.1",
                  "title": "Unit Rate",
                },
              ],
              "id": 1,
              "tagTitle": "6.RP.A.1",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="discussions-heading"
>
          Discussions
</h2>,
        Array [
          <DiscussionCard
            comments={
                        Array [
                                    Object {
                                      "comments": Array [
                                        Object {
                                          "body": "Yes but it's important",
                                        },
                                        Object {
                                          "body": "Here is a link to the progression docs: LINK",
                                        },
                                      ],
                                    },
                                  ]
            }
            discussion={
                        Object {
                                    "discussions": Array [
                                      Object {
                                        "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
                                        "tagTitle": "6.RP.A.1",
                                        "title": "Unit Rate",
                                      },
                                    ],
                                    "id": 1,
                                    "tagTitle": "6.RP.A.1",
                                  }
            }
/>,
        ],
      ],
      "className": "card-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Discussions",
          "className": "discussions-heading",
        },
        "ref": null,
        "rendered": "Discussions",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "comments": Array [
            Object {
              "comments": Array [
                Object {
                  "body": "Yes but it's important",
                },
                Object {
                  "body": "Here is a link to the progression docs: LINK",
                },
              ],
            },
          ],
          "discussion": Object {
            "discussions": Array [
              Object {
                "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
                "tagTitle": "6.RP.A.1",
                "title": "Unit Rate",
              },
            ],
            "id": 1,
            "tagTitle": "6.RP.A.1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="discussions-heading"
>
            Discussions
</h2>,
          Array [
            <DiscussionCard
              comments={
                            Array [
                                          Object {
                                            "comments": Array [
                                              Object {
                                                "body": "Yes but it's important",
                                              },
                                              Object {
                                                "body": "Here is a link to the progression docs: LINK",
                                              },
                                            ],
                                          },
                                        ]
              }
              discussion={
                            Object {
                                          "discussions": Array [
                                            Object {
                                              "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
                                              "tagTitle": "6.RP.A.1",
                                              "title": "Unit Rate",
                                            },
                                          ],
                                          "id": 1,
                                          "tagTitle": "6.RP.A.1",
                                        }
              }
/>,
          ],
        ],
        "className": "card-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Discussions",
            "className": "discussions-heading",
          },
          "ref": null,
          "rendered": "Discussions",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "comments": Array [
              Object {
                "comments": Array [
                  Object {
                    "body": "Yes but it's important",
                  },
                  Object {
                    "body": "Here is a link to the progression docs: LINK",
                  },
                ],
              },
            ],
            "discussion": Object {
              "discussions": Array [
                Object {
                  "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
                  "tagTitle": "6.RP.A.1",
                  "title": "Unit Rate",
                },
              ],
              "id": 1,
              "tagTitle": "6.RP.A.1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
